src/Compilers/CSharp/Test/Emit3/ExtensionTypeTests.cs

```diff
@@ -31312,4 +31435,742 @@ static void Main()
             //         i = d[1];
             Diagnostic(ErrorCode.ERR_NoTypeDef, "d[1]").WithArguments("Missing", "Missing, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null").WithLocation(12, 13));
     }
+
+    [Fact]
+    public void Obsolete_Property_Static()
+    {
+        var src = """
+object.Property1 = 1;
+_ = object.Property1;
+new object().Property2 = 1;
+_ = new object().Property2;
+
+implicit extension E for object
+{
+    [System.Obsolete("error", true)]
+    public static int Property1 { get => throw null; set => throw null; }
+
+    [System.Obsolete("error", true)]
+    public int Property2 { get => throw null; set => throw null; }
+}
+""";
+        var comp = CreateCompilation(src, targetFramework: TargetFramework.Net70);
+        comp.VerifyEmitDiagnostics(
+            // (1,1): error CS0619: 'E.Property1' is obsolete: 'error'
+            // object.Property1 = 1;
+            Diagnostic(ErrorCode.ERR_DeprecatedSymbolStr, "object.Property1").WithArguments("E.Property1", "error").WithLocation(1, 1),
+            // (2,5): error CS0619: 'E.Property1' is obsolete: 'error'
+            // _ = object.Property1;
+            Diagnostic(ErrorCode.ERR_DeprecatedSymbolStr, "object.Property1").WithArguments("E.Property1", "error").WithLocation(2, 5),
+            // (3,1): error CS0619: 'E.Property2' is obsolete: 'error'
+            // new object().Property2 = 1;
+            Diagnostic(ErrorCode.ERR_DeprecatedSymbolStr, "new object().Property2").WithArguments("E.Property2", "error").WithLocation(3, 1),
+            // (4,5): error CS0619: 'E.Property2' is obsolete: 'error'
+            // _ = new object().Property2;
+            Diagnostic(ErrorCode.ERR_DeprecatedSymbolStr, "new object().Property2").WithArguments("E.Property2", "error").WithLocation(4, 5));
+    }
+
+    [Fact]
+    public void ExtensionMemberLookup_Property_OnBase()
+    {
+        var src = """
+class C
```

Comment:
Please add a test for a scenario when `C` inherits from some other class (not an object).